packages/projection/src/operators/Mappers/withNftMetadata.ts

```diff
@@ -11,45 +11,36 @@ export interface WithNftMetadata {
 
 const KORALAB_NFT = 314;
 
-const parseOutputDatum = (outputs: Cardano.TxOut[]) => {
-  if (!outputs || outputs?.length === 0) return null;
-  return outputs
-    .filter(({ datum }) => !!datum)
-    .map(({ datum }) => {
-      const parsed = Asset.NftMetadata.fromPlutusData(datum, dummyLogger);
-      if (!parsed) return null;
-      return parsed;
-    });
-};
-
 export const withNftMetadata = unifiedProjectorOperator<WithMint & WithUtxo, WithNftMetadata>((evt) => {
-  const utxos = evt.utxo;
-  console.log('utxos', utxos);
-
-  return {
-    ...evt,
-    nftMetadata: new Map(
-      evt.mint
+  const handleProducedUtxo = () =>
+    evt.utxo.produced.flatMap(([, txOutUtxo]): Array<readonly [Cardano.AssetId, Asset.NftMetadata]> | null => {
+      const assets = txOutUtxo.value.assets;
+      if (!assets || assets.size === 0) return null;
+      const nftMetadata = Asset.NftMetadata.fromPlutusData(txOutUtxo.datum, dummyLogger);
+      if (!nftMetadata) return null;
+      return [...assets]
         .filter(
-          ({ assetName }) =>
-            !assetName || assetName?.label === Asset.AssetNameLabelNum.UserNFT || assetName?.label === KORALAB_NFT
+          ([, { assetName }]) => assetName.label === Asset.AssetNameLabelNum.UserNFT || assetName?.label === KORALAB_NFT
```

Comment:
I think the datum is stored together with the ReferenceNFT (not UserNFT). Please double-check in cip68

Reply:
Also we shouldn't hardcode KORALAB_NFT, but rather inject the labels via an argument (see example of such operator in filterProducedUtxoByAssetPolicyId)